sale_section_and_note_revamp/models/display_line_mixin.py

```diff
@@ -0,0 +1,177 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class DisplayLineMixin(models.AbstractModel):
+    _name = "display.line.mixin"
+    _description = """This model intends to ease the propagation and generation
+of sections and notes to any model in relation with a sale order line"""
+    _order = "sequence ASC, id ASC"
+
+    sequence = fields.Integer()
+    previous_line_id = fields.Many2one("sale.order.line")
+    next_line_id = fields.Many2one("sale.order.line")
+    display_type = fields.Char(store=False)
+
+    def get_section_or_note_class(self):
+        """Return the class names depending of the display type
+
+        Classes come from sale sections scss configuration
+        """
+        if self.is_section():
+            return "bg-200 font-weight-bold o_line_section"
+        elif self.is_note():
+            return "font-italic o_line_note"
+        return ""
+
+    def is_section(self):
+        """Quick method helper"""
+        self.ensure_one()
+        return self.display_type == "line_section"
+
+    def is_note(self):
+        """Quick method helper"""
+        self.ensure_one()
+        return self.display_type == "line_note"
+
+    def is_section_or_note(self):
+        """Quick method helper"""
+        self.ensure_one()
+        return self.is_section() or self.is_note()
+
+    def has_section(self):
+        """Quick method helper"""
+        return bool(self.get_section())
+
+    def has_note(self):
+        """Quick method helper"""
+        return bool(self.get_note())
+
+    def get_section(self):
+        """Returns the section of the current line"""
+        if not self.is_section():
+            previous_record = self.previous_line_id
+            while previous_record:
+                if previous_record.is_section():
+                    return previous_record
+                previous_record = previous_record.previous_line_id
+
+    def get_note(self):
```

Comment:
The method get_section loops in previous records to find the section.

I find it counter-intuitive that this method only checks if the next line is a note. I would expect this method to loop in previous records to find and get the related note, or a different naming